jerry-ext/include/jerryscript-ext/autorelease.h

```diff
@@ -0,0 +1,60 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef JERRYX_AUTORELEASE_H
+#define JERRYX_AUTORELEASE_H
+
+#include "autorelease.impl.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif /* __cplusplus */
+
+/*
+ * Macro for `const jerry_value_t` for which jerry_release_value() is
+ * automatically called when the variable goes out of scope.
+ *
+ * Example usage:
+ *
+ * ... {
+ *   JERRYX_VAL s = jerry_create_string(...);
+ *   // ...
+ *   // jerry_release_value(s) is called automatically when `s` goes out
+ *   // of scope.
+ * }
+ */
+#define JERRYX_VAL __JERRYX_VAL_IMPL
```

Comment:
let's not pollute the `jerryx_` "global" namespace. `JERRYX_VAL` sounds a bit like a catch-all term. `jerryx_auto_` `jerryx_ar_` `jerryx_autorelease_` would all be better. and one more note: this macro really acts like a type, it is the "smarter" variant of `jerry_value_t`. would the name `JERRYX_AR_VALUE_T` not align better with existing identifier names?